cartography/intel/okta/groups.py

```diff
@@ -1,298 +1,390 @@
-# Okta intel module - Group
-import json
+# Okta intel module - Groups
+import asyncio
 import logging
+import json
 from typing import Dict
 from typing import List
-from typing import Tuple
+from typing import Any
 
 import neo4j
-from okta.framework.ApiClient import ApiClient
-from okta.framework.OktaError import OktaError
-from okta.framework.PagedResults import PagedResults
-from okta.models.usergroup import UserGroup
-
-from cartography.intel.okta.sync_state import OktaSyncState
-from cartography.intel.okta.utils import check_rate_limit
-from cartography.intel.okta.utils import create_api_client
-from cartography.intel.okta.utils import is_last_page
+
+from cartography.client.core.tx import load
+from cartography.graph.job import GraphJob
+from okta.client import Client as OktaClient
+from okta.models.group import Group as OktaGroup
+from okta.models.group_rule import GroupRule as OktaGroupRule
+from okta.models.role import Role as OktaGroupRole
+from okta.models.user import User as OktaUser
+from cartography.models.okta.group import OktaGroupSchema
+from cartography.models.okta.group import OktaGroupRuleSchema
+from cartography.models.okta.group import OktaGroupRoleSchema
 from cartography.util import timeit
 
+
 logger = logging.getLogger(__name__)
 
+####
+# Groups
+####
+
 
 @timeit
-def _get_okta_groups(api_client: ApiClient) -> List[str]:
+def sync_okta_groups(
```

Comment:
This function syncs group roles, groups, and then group rules. It isn't nested deeply so I think it'd be straightforward to split to 3 different functions: sync_group_roles(), sync_groups(), and sync_group_rules(). I think the end result will be easier to read.

Also, is the ordering correct -- should we instead make it groups, then group roles, then group rules?

Reply:
Update: This is a nonblocker.